addons/website_slides_survey/controllers/main.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request
+from odoo.osv import expression
+from odoo.addons.website_profile.controllers.main import WebsiteProfile
+
+
+class WebsiteSlidesSurvey(WebsiteProfile):
+    def _prepare_user_profile_values(self, user, **kwargs):
+        """Loads all data required to display the certification attempts of the given user"""
+        values = super(WebsiteSlidesSurvey, self)._prepare_user_profile_values(user, **kwargs)
+        values['show_certification_tab'] = ('user' in values) and (
+            values['user'].id == request.env.user.id or \
+            request.env.user.has_group('survey.group_survey_manager')
+        )
+
+        if not values['show_certification_tab']:
+            return values
+
+        domain = [('survey_id.certification', '=', True)]
+        if values['user'].mapped('email'):
+            domain = expression.AND([domain, ['|',
+                ('email', 'in', values['user'].mapped('email')),
+                ('partner_id', 'in', values['user'].mapped('partner_id.id'))
+            ]])
+        else:
+            domain = expression.AND([domain, [
+                ('partner_id', 'in', values['user'].mapped('partner_id.id'))
+            ]])
+
+        if 'search' in kwargs:
+            values['active_tab'] = 'certification'
+            values['certification_search_terms'] = kwargs['search']
+            domain = expression.AND([domain,
+                [('survey_id.title', 'ilike', kwargs['search'])]
+            ])
+
+        certifications_attempts = []
+        attempt_number = 1
+        current_survey_id = False
+        UserInputSudo = request.env['survey.user_input'].sudo()
+
+        for user_input in UserInputSudo.search(domain, order='survey_id asc, create_date asc'):
+            survey_id = user_input.survey_id.id
+            if not current_survey_id or current_survey_id != survey_id:
+                attempt_number = 1
+                current_survey_id = survey_id
+            certifications_attempts.append({
+                'user_input': user_input,
+                'attempt_number': attempt_number
+            })
+            attempt_number += 1
+
+        order_by = kwargs.get('order', 'create_date')
+        if order_by == 'create_date':
+            certifications_attempts.sort(key=lambda e: e['user_input'].create_date, reverse=True) # inplace
```

Comment:
If the end user has no way to sort the results, I would hardcode the order in the search call and leave it like that to simplify.